### PR TITLE
CP-2_Reverse_String_Special

### PR DESCRIPTION
**Pull request by gyula-mezo**
- initial solution

### CODE REVIEW

File: src/main/java/com/epam/codingpractice/reversestringspecial/ReverseStringSpecial.java [hunk 1]
```diff
@@ -9,8 +14,32 @@
...
      */
-    public static String reverseSpecial(String str, char stayInPlaceCharacter){
-        throw new UnsupportedOperationException();
+    public static String reverseSpecial(String str, char stayInPlaceCharacter) {
+        if (str == null) {
+            throw new IllegalArgumentException("we don't deal with nulls");
+        }
+        char[] inputCharacters = str.toCharArray();
+
+        LinkedList<Character> characters = new LinkedList<>();
```
Comment (Zolikon) [inline src/main/java/com/epam/codingpractice/reversestringspecial/ReverseStringSpecial.java]: LinkedList is a good choice, but SOLI**D**

File: src/main/java/com/epam/codingpractice/reversestringspecial/ReverseStringSpecial.java [hunk 2]
```diff
@@ -9,8 +14,32 @@
...
+                characters.add(inputCharacters[i]);
+            }
+        }
+
+        stayInPlaceCharacterPositions.keySet().forEach(stayInPlaceCharacterPosition -> characters.add(stayInPlaceCharacterPosition, stayInPlaceCharacter));
+
+        return joinCharacters(characters);
+    }
+
+    private static String joinCharacters(LinkedList<Character> characters) {
```
Comment (Zolikon) [inline src/main/java/com/epam/codingpractice/reversestringspecial/ReverseStringSpecial.java]: same here

File: src/main/java/com/epam/codingpractice/reversestringspecial/ReverseStringSpecial.java (hunk 1 above)
Comment (gyula-mezo) [reply src/main/java/com/epam/codingpractice/reversestringspecial/ReverseStringSpecial.java]: Fixed. 👍

File: src/main/java/com/epam/codingpractice/reversestringspecial/ReverseStringSpecial.java (hunk 2 above)
Comment (gyula-mezo) [reply src/main/java/com/epam/codingpractice/reversestringspecial/ReverseStringSpecial.java]: Fixed 👍

**Pull request closed without merging**